`Could not find module ‘Prelude’` on macOS

<!--
Thanks for reporting your issue.
Please fill out the sections below.
-->

- [x] I read [the conda-forge documentation](https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge) and could not find the solution for my problem there.

Issue:

When trying to compile a simple hello world program, am running into some issues.

```Haskell
-- hello.rs

module Main where

main = putStrLn "Hello World"
```

Am running this to compile and am getting the error below.

```bash
$ ghc hello.hs -o hello
```

```Haskell
[1 of 1] Compiling Main             ( hello.hs, hello.o )

hello.hs:1:8: error:
    Could not find module ‘Prelude’
    There are files missing in the ‘base-4.14.1.0’ package,
    try running 'ghc-pkg check'.
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
  |
1 | module Main where
  |        ^^^^
```

Tried running `ghc-pkg check` and got a bunch of warnings. Running `ghc-pkg recache` as suggested doesn't seem to fix the issue.

```
$ ghc-pkg check
WARNING: cache is out of date: /Users/jkirkham/miniconda/envs/ghc/lib/ghc-8.10.4/package.conf.d/package.cache
ghc will see an old view of this package db. Use 'ghc-pkg recache' to fix.
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/directory-1.3.6.0/directory.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/terminfo-0.4.1.4/terminfo.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/stm-2.5.0.0/stm.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/process-1.6.9.0/process.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/time-1.9.3/time.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/filepath-1.4.2.1/filepath.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/libiserv-8.10.4/libiserv.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/pretty-1.1.3.6/pretty.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/integer-gmp-1.0.3.0/integer-gmp.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/mtl-2.2.2/mtl.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/unix-2.7.2.2/unix.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/ghci-8.10.4/ghci.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/hpc-0.6.1.0/hpc.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/template-haskell-2.16.0.0/template-haskell.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/ghc-boot-8.10.4/ghc-boot.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/ghc-prim-0.6.1/ghc-prim.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/base-4.14.1.0/base.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/parsec-3.1.14.0/parsec.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/ghc-heap-8.10.4/ghc-heap.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/ghc-boot-th-8.10.4/ghc-boot-th.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/ghc-compact-0.1.0.0/ghc-compact.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/containers-0.6.2.1/containers.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/deepseq-1.4.4.0/deepseq.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/array-0.5.4.0/array.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/text-1.2.4.1/text.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/Cabal-3.2.1.0/Cabal.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/ghc-8.10.4/ghc.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/binary-0.8.8.0/binary.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/transformers-0.5.6.2/transformers.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/exceptions-0.10.4/exceptions.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/bytestring-0.10.12.0/bytestring.haddock doesn't exist or isn't a file
Warning: haddock-interfaces: /Users/jkirkham/miniconda/envs/ghc/share/doc/ghc-8.10.4/html/libraries/haskeline-0.8.0.1/haskeline.haddock doesn't exist or isn't a file
```

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list
# packages in environment at /Users/jkirkham/miniconda/envs/ghc:
#
# Name                    Version                   Build  Channel
c-compiler                1.3.0                h0d85af4_0    conda-forge
cctools                   949.0.1             hd9211c8_21    conda-forge
cctools_osx-64            949.0.1             h6407bdd_21    conda-forge
clang                     11.1.0               h694c41f_1    conda-forge
clang-11                  11.1.0          default_he082bbe_1    conda-forge
clang_osx-64              11.1.0               hb91bd55_3    conda-forge
clangxx                   11.1.0          default_he082bbe_1    conda-forge
compiler-rt               11.1.0               h654b07c_0    conda-forge
compiler-rt_osx-64        11.1.0               h8c5fa43_0    conda-forge
ghc                       8.10.4               h695349b_0    conda-forge
gmp                       6.2.1                h2e338ed_0    conda-forge
ld64                      530                 hd2e7500_21    conda-forge
ld64_osx-64               530                 he8994da_21    conda-forge
ldid                      2.1.2                h7660a38_2    conda-forge
libclang-cpp11.1          11.1.0          default_he082bbe_1    conda-forge
libcxx                    12.0.1               habf9029_0    conda-forge
libffi                    3.3                  h046ec9c_2    conda-forge
libllvm11                 11.1.0               hd011deb_2    conda-forge
llvm-openmp               12.0.1               hda6cdc1_1    conda-forge
llvm-tools                11.1.0               hd011deb_2    conda-forge
ncurses                   6.2                  h2e338ed_4    conda-forge
tapi                      1100.0.11            h9ce4665_0    conda-forge
xz                        5.2.5                haf1e3a3_1    conda-forge
zlib                      1.2.11            h7795811_1010    conda-forge
```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info

     active environment : ghc
    active env location : /Users/jkirkham/miniconda/envs/ghc
            shell level : 2
       user config file : /Users/jkirkham/.condarc
 populated config files : /Users/jkirkham/.condarc
          conda version : 4.10.3
    conda-build version : 3.21.4
         python version : 3.8.10.final.0
       virtual packages : __osx=10.15.7=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /Users/jkirkham/miniconda  (writable)
      conda av data dir : /Users/jkirkham/miniconda/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/conda-forge/osx-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/jkirkham/miniconda/pkgs
                          /Users/jkirkham/.conda/pkgs
       envs directories : /Users/jkirkham/miniconda/envs
                          /Users/jkirkham/.conda/envs
               platform : osx-64
             user-agent : conda/4.10.3 requests/2.26.0 CPython/3.8.10 Darwin/19.6.0 OSX/10.15.7
                UID:GID : 523832026:1019530959
             netrc file : None
           offline mode : False
```
</details>

--- a/recipe/activate.sh
+++ b/recipe/activate.sh
@@ -1,0 +1,3 @@
+#!/bin/bash
+
+ghc-pkg recache

--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -47,7 +47,10 @@ pushd source
   # stage1 compiler: --build=$GHC_BUILD --host=$GHC_BUILD --target=$GHC_HOST
   # stage2 compiler: --build=$GHC_BUILD --host=$GHC_HOST --target=$GHC_HOST
   if [[ "${target_platform}" == linux-* ]]; then
-    export CC=$GCC
+    export CC=$(basename $GCC)
+    export AR=$(basename $AR)
+    export LD=$(basename $LD)
+    export RANLIB=$(basename $RANLIB)
   fi
   cp $BUILD_PREFIX/share/gnuconfig/config.* .
   (

--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -60,3 +60,7 @@ pushd source
   find $PREFIX/lib/ghc-${PKG_VERSION} -name '*_p.a' -delete
   find $PREFIX/lib/ghc-${PKG_VERSION} -name '*.p_o' -delete
 popd
+
+if [[ "${target_platform}" == "${build_platform}" ]]; then
+  ghc-pkg recache
+fi

--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -64,6 +64,9 @@ pushd source
   find $PREFIX/lib/ghc-${PKG_VERSION} -name '*.p_o' -delete
 popd
 
-if [[ "${target_platform}" == "${build_platform}" ]]; then
-  ghc-pkg recache
-fi
+# Delete package cache as it is invalid on installation.
+# This needs to be regenerated on activation.
+rm $PREFIX/lib/ghc-${PKG_VERSION}/package.conf.d/package.cache
+
+mkdir -p "${PREFIX}/etc/conda/activate.d"
+cp "${RECIPE_DIR}/activate.sh" "${PREFIX}/etc/conda/activate.d/${PKG_NAME}_activate.sh"

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -17,7 +17,7 @@ source:
     sha256: 52af871b4e08550257d720c2944ac85727d0b948407cef1bebfe7508c224910e
 
 build:
-  number: 1
+  number: 2
   skip: True  # [not unix]
 
 requirements:
@@ -40,6 +40,8 @@ requirements:
     - ncurses
     - libffi
     - cffi
+  run:
+    - {{ c_compiler }}_{{ target_platform }} >={{ c_compiler_version }}
 
 test:
   files:

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -33,7 +33,7 @@ requirements:
     - automake
     - autoconf
     - libtool
-    - llvmdev  # [linux]
+    - llvmdev 9.*  # [aarch64]
   host:
     - xz
     - gmp
@@ -42,6 +42,7 @@ requirements:
     - cffi
   run:
     - {{ c_compiler }}_{{ target_platform }} >={{ c_compiler_version }}
+    - llvm-tools 9.*  # [aarch64]
 
 test:
   files:

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -33,7 +33,7 @@ requirements:
     - automake
     - autoconf
     - libtool
-    - llvmdev  # [aarch64]
+    - llvmdev  # [linux]
   host:
     - xz
     - gmp

